### PR TITLE
Ansible playbook to configure sshd for Conch CA certs.

### PR DESCRIPTION
**Pull request by MaxBed4d**

### CODE REVIEW

File: ansible/ca-cert.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# An ansible playbook to configure the SSHD configuration to enable CA cert auth for SSH.
+# Remember to export CI_CLOUD if it isn't part of your environment's variables.
```
Comment (sd109) [inline ansible/ca-cert.yml:2]: Why is this needed? How are you running this playbook?

File: ansible/ca-cert.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+# NOTE: Change the src for the `ssh_signing_key.pub` to be your corresponding directory.
+
+- hosts: login
+  gather_facts: true
+  become: true
+  tasks:
+    - name: Copy ssh public key
+      ansible.builtin.copy:
+        src: /var/lib/rocky/conch/ssh_signing_key.pub
```
Comment (sjpb) [inline ansible/ca-cert.yml:12]: How's this getting configured? You just created this by hand right?

File: ansible/ca-cert.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# An ansible playbook to configure the SSHD configuration to enable CA cert auth for SSH.
+# Remember to export CI_CLOUD if it isn't part of your environment's variables.
+
+# NOTE: Change the src for the `ssh_signing_key.pub` to be your corresponding directory.
+
+- hosts: login
+  gather_facts: true
```
Comment (sjpb) [inline ansible/ca-cert.yml:7]: Doesn't look like you need facts, so I'd turn it off or at least leave it out (which will default to on, but at least you're not documenting that you need something you don't).

File: ansible/ca-cert.yml (hunk 1 above)
Comment (sjpb) [inline ansible/ca-cert.yml:2]: CI_CLOUD is only required if you're using app creds. Playbooks aren't the place to document the wriggles of authing to openstack TBH.

Comment (sjpb) [reply ansible/ca-cert.yml:2]: Yeah we need to talk about where this is included in site.yml, there's a bigger job/discussion here than just adding a playbook I think.

File: ansible/ca-cert.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+      ansible.builtin.copy:
+        src: /var/lib/rocky/conch/ssh_signing_key.pub
+        dest: /etc/ssh/ca_user_key.pub
+        owner: root
+        group: root
+        mode: '0644'
+        remote_src: true
+
+    - name: Ensure CA Certs are accepted
+      ansible.builtin.lineinfile:
```
Comment (sjpb) [inline ansible/ca-cert.yml:20]: Does this need a `regexp:` line to make it idempotent? E.g. if its already there, but commented out?

File: ansible/ca-cert.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        line: 'TrustedUserCAKeys /etc/ssh/ca_user_key.pub'
+        dest: /etc/ssh/sshd_config
+
+    - name: Restart SSH service
+      ansible.builtin.systemd:
+        name: sshd
+        state: restarted
```
Comment (sjpb) [inline ansible/ca-cert.yml:27]: Depending on how this should actually be integrated, this might need gating somehow

File: ansible/ca-cert.yml [hunk 6]
```diff
@@ -0,0 +1,27 @@
+# An ansible playbook to configure the SSHD configuration to enable CA cert auth for SSH.
+# Remember to export CI_CLOUD if it isn't part of your environment's variables.
+
+# NOTE: Change the src for the `ssh_signing_key.pub` to be your corresponding directory.
+
+- hosts: login
```
Comment (sjpb) [inline ansible/ca-cert.yml:6]: Normally we'd have a group for the feature, but I'm not clear on which feature this actually belongs to. I suggested login to @MaxBed4d understanding this was a one-off play running against an already-deployed cluster. But that wouldn't be a PR ...

**Comment by sd109** [comment]
As discussed elsewhere, let's just merge this as-is for now since it's targeting the Waldur proof of concept branch. We can worry about polishing things later if needed.

**Pull request merged**